### PR TITLE
Added basic www tests: Jest+RTL

### LINKED ISSUE #22: Set up basic frontend (smoke) tests
* Commit a static package JSON into the repo that demonstrates all the weirdness we care about.
  * Ideally by including such a "fake" package in the source tree, and generating the JSON from that, so that we can use the same data to test the indexing part.
* Add very basic tests that verify all the various pages even render at all.
  * https://nextjs.org/docs/testing shows Next.js supports 4 testing frameworks, I only know one of them, needs research or input?
* Add somewhat less basic tests that verify the correct content is showing up (like, correct modules in the sidebar, correct items in the table, correct function signatures in detail pages)
  * This is possibly snapshot testing, though the design / layout is bound to change a whole lot and quickly, so maybe it's too early for that.
* Run all this in CI, obviously

### PR DESCRIPTION
**Pull request by abesto**
Things to note:
* `npm run test` starts a process that watches for code changes and executes tests as appropriate.
* Project JSON is committed into the repo; needs manual updates currently.
* Test methodology: mock out as little as possible, render the top-level page components.
* What's mocked for tests:
  * A server that returns the project JSON
  * The path where project JSONs are looked up is overridden with an ENV var (in `www/.env.test`)
* Large tests because in the words of the dad of RTL: https://kentcdodds.com/blog/write-fewer-longer-tests (and indeed trying to make them smaller got real painful real fast)

There's much more that could be tested; things that immediately come to mind:
* Navigation between SPA pages, verifying that there is no page reload
* Snapshot tests (but maybe it's still too early for those)

That said, this should give us the confidence we need that changes don't COMPLETELY mess up the site.

Closes #22

### CODE REVIEW

**Review by zsol — APPROVED** [review]
LGTM, I'll wait for @anathien to take a look before merging

File: www/__tests__/class.test.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        `/project-alpha/alpha.core/${sym}`
+      );
+    }
+
+    // Class declaration
+    getByText(
+      (_, element) =>
+        element?.nodeName === "PRE" &&
+        element?.textContent === "class alpha.core.Helper()"
+    );
```
Comment (anathien) [inline www/__tests__/class.test.tsx:42]: What do these do? They don't look like being tests (there are no asserts) and they don't look to be utils (not used anywhere in the test suite)... Please teach me 🙂

File: www/__tests__/index.test.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  it("contains link to project-alpha", async () => {
+    const propsResult = await getStaticProps({});
+    expect(propsResult).toHaveProperty("props");
+    const props = (propsResult as { props: Props }).props;
+
+    const { getByRole } = render(<Index {...props} />);
+    const link = getByRole("link", { name: "project-alpha" });
+    expect(link).toBeInTheDocument();
+    expect(link).toHaveAttribute("href", "/project-alpha");
+    expect(link.parentElement).toHaveTextContent("project-alpha (0)");
```
Comment (anathien) [inline www/__tests__/index.test.tsx:16]: 👌

File: www/__tests__/project.test.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    // Modules table
+    const modulesHeading = getByRole("heading", { name: "Modules" });
+    const modulesTable = modulesHeading.nextElementSibling as HTMLTableElement;
+    expect(modulesTable).not.toBeNull();
+    expect(modulesTable.nodeName).toBe("TABLE");
+    for (const mod of ["alpha", "alpha.core", "alpha.foo"]) {
+      expect(
+        within(modulesTable).getByRole("link", { name: mod })
+      ).toHaveAttribute("href", `/project-alpha/${mod}`);
+    }
```
Comment (anathien) [inline www/__tests__/project.test.tsx:25]: 👌

**Review by anathien — APPROVED** [review]
Very nice! 🙂 

As a rule of thumb, I prefer smaller test cases, so that we can narrow the cause of breakages a bit. Although I agree with Mr Dodds that very granular test cases are a pain, I have found moderate sized ones quite helpful. Can take a look at breaking things down a bit once we have our ducks in order.

File: www/__tests__/class.test.tsx (hunk 1 above)
Comment (abesto) [reply www/__tests__/class.test.tsx:42]: The `get*` queries explode if there are no matches; so these are kinda sorta tests that verify such an element exists.

Comment (abesto) [reply www/__tests__/class.test.tsx:42]: ... I guess one could instead do `expect(findByRole(...)).to.notBeNull()`, matter of taste IMHO.

**Comment by zsol** [comment]
🎊

**Pull request merged**